operator-k8s/src/controller/cluster/mod.rs

```diff
@@ -1,7 +1,81 @@
+use clippy_utilities::NumericCast;
+use prometheus::{Error, Histogram, HistogramOpts, HistogramTimer, IntCounterVec, Opts, Registry};
+
+use std::iter::repeat;
+use std::ops::Mul;
+
+use crate::controller::Metrics;
+
 /// Controller v1alpha
 mod v1alpha;
 /// Controller v1alpha1
 mod v1alpha1;
 
 /// Current controller of cluster
 pub(crate) type Controller = v1alpha::ClusterController;
+
+/// Cluster metrics
+pub(crate) struct ClusterMetrics {
+    /// Reconcile duration histogram
+    reconcile_duration: Histogram,
+    /// Reconcile failed count
+    reconcile_failed_count: IntCounterVec,
+}
+
+impl Default for ClusterMetrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Metrics for ClusterMetrics {
+    /// Register metrics
+    fn register(&self, registry: &Registry) -> Result<(), Error> {
+        registry.register(Box::new(self.reconcile_duration.clone()))?;
+        registry.register(Box::new(self.reconcile_failed_count.clone()))
+    }
+
+    /// Record duration
+    fn record_duration(&self) -> HistogramTimer {
+        self.reconcile_duration.start_timer()
```

Comment:
Does this has a `stop timer` method that need to be called?

Reply:
`stop timer` will be invoked when HistogramTimer dropped.